server/analyzer/comparison_casts.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2024 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package analyzer
+
+import (
+	"fmt"
+
+	"github.com/dolthub/go-mysql-server/sql"
+	"github.com/dolthub/go-mysql-server/sql/analyzer"
+	"github.com/dolthub/go-mysql-server/sql/expression"
+	"github.com/dolthub/go-mysql-server/sql/plan"
+	"github.com/dolthub/go-mysql-server/sql/transform"
+
+	pgexprs "github.com/dolthub/doltgresql/server/expression"
+	"github.com/dolthub/doltgresql/server/functions/framework"
+	pgtypes "github.com/dolthub/doltgresql/server/types"
+)
+
+// ComparisonCasts handles implicit conversions for comparisons. We cannot yet implement the comparison operators as GMS
+// relies on specific expressions some cases. Eventually, we'll need to decouple the dependency, so that we may use the
+// appropriate functions. This is technically incorrect, but should let us make progress until then.
+func ComparisonCasts(ctx *sql.Context, a *analyzer.Analyzer, node sql.Node, scope *plan.Scope, selector analyzer.RuleSelector) (sql.Node, transform.TreeIdentity, error) {
+	if disjointedNode, ok := node.(plan.DisjointedChildrenNode); ok {
+		return handleDisjointedNodes(ctx, a, disjointedNode, scope, selector, ComparisonCasts)
```

Comment:
Why doesn't `NodeExprs` and friends do this automatically?

Reply:
Or another function in the transform package for these nodes. This kind of thing should be handled on the engine side, not here.